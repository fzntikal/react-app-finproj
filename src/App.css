::-webkit-scrollbar {
  width: 5px;
}
::-webkit-scrollbar-track {
  background: #f1f1f1;
}
::-webkit-scrollbar-thumb {
  background: black;
}



button {
  border: none;
  padding: 8px;
  border-radius: 10px;
  background-color: rgba(100,100,100,0.3);
  cursor: pointer;
}

/* Navbar Header */
.navhead {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0;
  padding: 0;
  cursor: default;
}
.logo {
  width: 60px;
  height: 55px;
}
.navbar {
  border-bottom: solid 1px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0;
  padding: 0 20px;
}

ul {
  list-style-type: none;
  padding: 0;
  display: inline-flex;
  flex-direction: row;
  justify-content: flex-end;
}

li {
  margin-left: 2px;
  padding: 5px;
  border-radius: 10px;
  /* border: solid 1px; */
}

.nav-link {
  text-decoration: none;
  color: black;
  border: none;
}

li:hover {
  animation: nav-anim 1s forwards ;
}

@keyframes nav-anim {
  100% {
    background-color: rgba(200,200,200,0.7);
  }
}


/* Home Page */
.home {
  text-align: center;
  cursor: default;
  padding-bottom: 50px;
}
.home::before {
  content: '';
  background-image: url('./icons/tower.png');
  background-size: 200px;
  background-repeat: no-repeat;
  z-index: -1;
  position: fixed;
  top: 60%;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0.1;
}

.home-desc {
  margin: 0 30px;
  padding-bottom: 20px;
}
.home-detail {
  display: flex;
  flex-direction: column;
  margin: 20px 50px;
  padding: 15px;
  border: solid 2px;
  border-radius: 10px;
}

/* Contact Page */
.contactForm {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  padding-bottom: 50px;
}
.contactForm #name, #email, textarea {
  text-align: center;
  width: 250px;
}

.contactForm::before {
  content: '';
  background-image: url('./icons/knight.png');
  background-size: 200px;
  background-repeat: no-repeat;
  z-index: -1;
  position: fixed;
  top: 60%;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0.1;
}

.contactForm label, button{
  margin-top: 25px;
  font-weight: bold;
}



/* Products */
.products {
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  margin: 0 100px;
  padding-bottom: 50px;
}
.products::before {
  content: '';
  background-image: url('./icons/helmet.png');
  background-size: 145px;
  background-repeat: no-repeat;
  z-index: -1;
  position: fixed;
  top: 60%;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0.1;
}

.search-bar {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.search-bar input {
  text-align: center;
  border-radius: 10px;
}

.product-table {
  margin: 0 50px;
}

.product-table th {
  border-bottom: solid 1px;
  border-left: solid;
  padding-bottom: 5px;
  background: rgba(200,200,200,0.5);
}
.product-table td {
  border: solid 1px;
  padding: 5px;
}


/* Extras */
.extras {
  position: fixed;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

#stars {
  position: fixed;
  right: 0;
  left: 0;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}
.ex-star1 {
  height: 20px;
  position: relative;
  width: 20px;
  border-radius: 50%;
  background: white;
  animation: twinkle 2s infinite;
}
.ex-star2 {
  height: 10px;
  width: 10px;
  position: relative;
  border-radius: 50%;
  background: white;
  animation: twinkle 4s infinite;
}
.ex-star3 {
  height: 30px;
  width: 30px;
  position: relative;
  border-radius: 50%;
  background: white;
  animation: twinkle 3s infinite;
}
.ex-star4 {
  height: 20px;
  width: 20px;
  position: relative;
  border-radius: 50%;
  background: white;
  animation: twinkle 2.3s infinite;
}
@keyframes twinkle {
  0% {
    top: 150px;
    opacity: 0;
  }
  50% {
    top: 300px;
    opacity: 1;
  }
  100% {
    top: 150px;
    opacity: 0;
  }
}

/* Footer */

.footer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  width: 100%;
  height: 40px;
  bottom: 0;
  background-color: rgba(200,200,200,0.9);
  font-size: 14px;
}

.footer-links a img {
  width: auto;
  height: 25px;
}