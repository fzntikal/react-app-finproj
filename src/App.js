import React from 'react';
import Website from './pages/website';
import './App.css';

/**
 * Konsep utama dalam website ini adalah Online Shop.
 * Pada halaman Home, terdapat informasi mengenai toko dan barangnya.
 *  code dapat diubah dengan mudah untuk menambah/mengurangi informasi yang ada.
 * 
 * Pada halaman Contact Us, terdapat sebuah form.
 *  Form ini digunakan agar customer dapat menghubungi email toko.
 * 
 * Pada halaman Products, customer dapat melihat tabel produk.
 *  Terdapat fitur filter dan search sederhana yang bisa digunakan.
 * 
 * Pada halaman EXTRA, hanya terdapat sebuah tombol.
 *  Halaman ini dibuat di luar konteks Online shop,
 *  dan hanya berisikan tombol untuk merubah warna latar,
 *  serta menampilkan animasi sederhana.
 */

export default function App() {
    return(
        <Website /> // menampung keseluruhan webpages
    )
}

