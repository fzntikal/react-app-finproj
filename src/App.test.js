import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders stuff', () => {
  const { getByText }  = render(<App />);
  const linkElement = getByText(/welcome/ig);
  expect(linkElement).toBeInTheDocument();
});