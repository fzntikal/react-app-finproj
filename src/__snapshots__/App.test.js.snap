// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change theme 1`] = `
Array [
  <div
    className="navbar"
  >
    <h1
      className="navhead"
    >
      AshenShop
    </h1>
    <img
      alt="logo"
      className="logo"
      src="https://i.pinimg.com/originals/ed/1c/6e/ed1c6ec431f5f0a6edfb108aab61b97f.png"
    />
    <ul>
      <li>
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/contactUs"
          onClick={[Function]}
        >
          Contact Us
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/products"
          onClick={[Function]}
        >
          Products
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/extras"
          onClick={[Function]}
        >
          EXTRA
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="home"
  >
    <h2>
      Welcome to this page, Sire!
    </h2>
    <h3>
      <em>
        "Praise the sun!"
      </em>
       -Solaire of Astora
    </h3>
    <p>
      Do you like medieval stuffs? We got it! Enjoy your time here with us! 
      <br />
      We provide the best weapons, armors, and other medieval stuff, feel free to request for new items.
    </p>
    <div
      className="home-desc"
    >
      <h2>
        The stuff currently in our possession:
      </h2>
      <ul
        className="home-detail"
      >
        <li>
          Sword
          :
          <br />
          A melee weapon that are used mainly for side-arms, nobility often decorate their personal swords to fit their noble status
        </li>
        <li>
          Spear
          :
          <br />
          Long-range melee weapon, perfect to use against cavalry and to hold enemy back. Majority of low ranked foot soldiers are wielding this weapon.
        </li>
        <li>
          Shield
          :
          <br />
          A defensive tools made to protect its wielder.
        </li>
        <li>
          Armor
          :
          <br />
          Basically the things medieval knight wears. Plates, leathers, chainmails, you name it! All made to protect their body.
        </li>
      </ul>
      <h2>
        And many more medieval stuff!
      </h2>
      <div>
        <p>
          Click these buttons below to see illustration of the medieval stuff!
        </p>
        <button
          onClick={[Function]}
        >
          Sword
        </button>
        <button
          onClick={[Function]}
        >
          Spear
        </button>
        <button
          onClick={[Function]}
        >
          Shield
        </button>
        <button
          onClick={[Function]}
        >
          Armor
        </button>
        <p>
          
        </p>
      </div>
       
    </div>
  </div>,
  <footer
    className="footer"
  >
    <div
      className="footer-links"
    >
      <a
        href="https://www.facebook.com/felzanova"
      >
        <img
          alt="fb-icon"
          src="fb-icon.png"
        />
      </a>
      <a
        href="https://twitter.com/tiikall"
      >
        <img
          alt="tw-icon"
          src="tw-icon.png"
        />
      </a>
      <a
        href="https://www.instagram.com/ashencowl/"
      >
        <img
          alt="ig-icon"
          src="ig-icon.png"
        />
      </a>
    </div>
    <p>
      Site made by Fauzan Tikal for study purposes.
    </p>
  </footer>,
]
`;
